boltdesignsystem/bolt
# z-index | Tools | Map, function, and mixin refactor

## Jira

http://vjira2:8080/browse/BDS-659

## Summary

Refactored z-index settings and tools in `bolt-core`.

## Details

1. New map is created using auto index method
2. Function and mixin usage stay the same, deprecation warnings are in place for all deprecated z-index names
3. All existing z-index usage has been converted to use the new z-index names
4. Doc page is revamped to clearly document how to use the util classes and the function
5. A decision tree graphic has been made to help understand the complexity of z-index

## How to test

Run the branch locally and go through the documentation page (http://localhost:3000/pattern-lab/?p=visual-styles-z-index). Create a dummy page and some dummy CSS to test out the `bolt-z-index()` function, make sure the function is working as expected according to the stacking order specified in the map.
